src/commands/test/test_collector_test.py

```diff
@@ -0,0 +1,185 @@
+    )
+
+    assert_tested_file_names(result.test_subjects, ["test_foo.cairo"])
+    assert result.test_cases_count == 2
+
+
+def test_breakage_upon_broken_test_file(starknet_compiler, project_root):
+    test_collector = TestCollector(starknet_compiler)
+    cast(MagicMock, starknet_compiler.get_functions).side_effect = PreprocessorError("")
+
+    with pytest.raises(TestCollectingException):
+        test_collector.collect(target=project_root)
+
+
+def test_collecting_specific_file(starknet_compiler, project_root: Path):
+    test_collector = TestCollector(starknet_compiler)
+
+    result = test_collector.collect(project_root / "foo" / "test_foo.cairo")
+
+    assert_tested_file_names(result.test_subjects, ["test_foo.cairo"])
+
+
+def test_collecting_specific_function(starknet_compiler, project_root: Path):
+    test_collector = TestCollector(starknet_compiler)
+
+    result = test_collector.collect(project_root / "foo" / "test_foo.cairo::test_foo")
+
+    assert_tested_file_names(result.test_subjects, ["test_foo.cairo"])
+    assert result.test_cases_count == 1
+
+
+def test_logging_collected_one_test_suite_and_one_test_case(mocker: MockerFixture):
+    logger_mock = mocker.MagicMock()
+
+    TestCollector.Result(
+        test_subjects=[
+            TestSubject(
+                test_functions=[
+                    StarknetCompiler.AbiElement(
+                        name="foo", type="function", inputs=[], outputs=[]
+                    ),
+                ],
+                test_path=Path(),
+            )
+        ],
+        test_cases_count=1,
+    ).log(logger_mock)
+
+    cast(MagicMock, logger_mock.info).assert_called_once_with(
+        "Collected 1 suite, and 1 test case"
+    )
+
+
+def test_logging_many_test_suites_and_many_test_cases(mocker: MockerFixture):
+    logger_mock = mocker.MagicMock()
+
+    TestCollector.Result(
+        test_subjects=[
+            TestSubject(
+                test_functions=[
+                    StarknetCompiler.AbiElement(
+                        name="foo", type="function", inputs=[], outputs=[]
+                    ),
+                ],
+                test_path=Path(),
+            ),
+            TestSubject(
+                test_functions=[
+                    StarknetCompiler.AbiElement(
+                        name="foo", type="function", inputs=[], outputs=[]
+                    ),
+                ],
+                test_path=Path(),
+            ),
+        ],
+        test_cases_count=2,
+    ).log(logger_mock)
+
+    cast(MagicMock, logger_mock.info).assert_called_once_with(
+        "Collected 2 suites, and 2 test cases"
+    )
+
+
+def test_logging_no_cases_found(mocker: MockerFixture):
+    logger_mock = mocker.MagicMock()
+
+    TestCollector.Result(
+        test_subjects=[],
+        test_cases_count=0,
+    ).log(logger_mock)
+
+    cast(MagicMock, logger_mock.warn).assert_called_once_with("No cases found")
```

Comment:
test_logging_preprocessing_error is probably missing

Reply:
I'm not sure if collector handles this, but just wanted to mention I don't know what will happen here

Reply:
`test_breakage_upon_broken_test_file` checks if `TestCollectingException` is raised. Logging this error is not a responsibility of this module.